assets/app/scripts/services/applicationGenerator.js

```diff
@@ -231,8 +222,8 @@ angular.module("openshiftConsole")
           },
           source: {
             git: {
-              ref: sourceRef,
-              uri: sourceUrl
+              ref: input.buildConfig.gitRef || 'master',
```

Comment:
Might be weird to have part of the URL disappear after the field blurs, particularly if the ref field is hidden under the advanced options link.